README.md

```diff
@@ -1,2 +1,52 @@
-# aws-cloudwatch-firehose
-Forwards logs from cloudwatch to NewRelic through firehose
+# aws-unified-firehose
+Forwards logs from cloudwatch to NewRelic through AWS firehose
+
+## Features
+
+- Collects logs from AWS CloudWatch.
+- Forwards logs to NewRelic using AWS Firehose.
+- Allows users to attach custom attributes to the logs to make it easier to search, filter, analyze, and parse the logs
+- Scalable and reliable log forwarding.
+
+## Requirements
+
+- AWS CLI already configured with Administrator permission
+- [Docker installed](https://www.docker.com/community-edition)
+- SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+
+## Deployment
+
+To try this lambda out you can use the `sam` cli to deploy the CFT (`firehose-template.yml`). Make sure **aws is properly authenticated with an account of your choice**.
+
+
+#### CloudFormation Parameters
+
+- `NewRelicRegion` : Can either be `US` or `EU` depending on which endpoint to be used to push logs to New Relic
+- `LicenseKey`: Used when forwarding logs to New Relic
+- `LogGroupConfig` : String representation of JSON array of objects of your CloudWatch LogGroup(s) and respective filter (if applicable) to set the Lambda function trigger.
+  - Example : ```[{"LogGroupName":"group1"}, {"LogGroupName":"group2", "FilterPattern":"ERROR"},  {"LogGroupName":"group3", "FilterPattern":"INFO"}]```
+- `LoggingFirehoseStreamName` : Name of new Data Firehose Delivery Stream (must be unique per AWS account in the same AWS Region)
+- `LoggingS3BackupBucketName`: S3 Bucket Destination for failed events (must be globally unique across all AWS accounts in all AWS Regions within a partition)
+- `EnableCloudWatchLoggingForFirehose`: Can either be `true` or `false` to enable CloudWatch logging for the Firehose stream.
```

Comment:
add detail like: it can help in troubleshooting issues in pushing data through firehose stream. Also mention what if the default value for this and some other parameters where we have allowed values.

Reply:
Added default values.